remote/analyze_handler.py

```diff
@@ -0,0 +1,301 @@
+
+    with zipfile.ZipFile(analyze_output_zip_file_path, 'a') as output:
+        for root, dirs, files in os.walk(output_dir_path):
+            for file in files:
+                output.write(os.path.join(root, file))
+
+    REDIS_DATABASE.hincrby(analyze_id, 'completed_parts', 1)
+
+    parts = REDIS_DATABASE.hget(analyze_id, 'parts')
+    completed_parts = REDIS_DATABASE.hget(analyze_id, 'completed_parts')
+
+    if parts == completed_parts:
+        REDIS_DATABASE.hset(analyze_id, 'state',
+                            AnalyzeStatus.ANALYZE_COMPLETED.name)
+        LOG.info('Analyze %s is completed.', analyze_id)
+
+
+def main():
+    parser = argparse.ArgumentParser(description=".....")
+
+    parser.add_argument('-w', '--workspace', type=str, dest='workspace',
+                        default='/workspace', help="...")
+
+    parser.add_argument('--no-cache', dest='use_cache',
+                        default=True, action='store_false')
+
+    args = parser.parse_args()
+
+    scheduler = sched.scheduler(time.time, time.sleep)
+
+    def check_queue():
+        scheduler.enter(10, 1, check_queue, ())
+        LOG.info('Check task queue in Redis.')
+        tasks = REDIS_DATABASE.lrange('ANALYSES_QUEUE', 0, -1)
+
+        task = REDIS_DATABASE.lpop('ANALYSES_QUEUE')
+
+        if task is not None:
+            LOG.info(
+                'Got a task: %s, starting analyze it, leftover task(s) %s', task, len(tasks))
+            analyze_id, part_number = str(task).split('-')
+
+            try:
+                analyze_dir_path, build_command, file_to_analyze_path = pre_analyze(
+                    analyze_id, part_number, args.workspace, args.use_cache)
+            except Exception:
+                REDIS_DATABASE.hset(analyze_id, 'state',
+                                    AnalyzeStatus.PREANALYZE_FAILED.name)
+                LOG.error('Failed pre-analyze step to read in file path.')
+
+            try:
+                analyze(analyze_id, analyze_dir_path,
+                        build_command, file_to_analyze_path)
+            except Exception:
+                REDIS_DATABASE.hset(analyze_id, 'state',
+                                    AnalyzeStatus.ANALYZE_FAILED.name)
+                LOG.error('Failed to read in file path.')
+
+            try:
+                post_analyze(analyze_id, part_number, args.workspace)
+            except Exception:
+                REDIS_DATABASE.hset(analyze_id, 'state',
+                                    AnalyzeStatus.PREANALYZE_FAILED.name)
+                LOG.error('Failed to read in file path.')
+
+        else:
+            LOG.info('No task.')
+
+    def schedule():
+        # check analyses queue in every 10 seconds
+        scheduler.enter(10, 1, check_queue, ())
+        scheduler.run()
+
+    schedule()
+
+
+if __name__ == "__main__":
+    REDIS_DATABASE = redis.Redis(
+        host='redis', port=6379, db=0, charset="utf-8", decode_responses=True)
```

Comment:
This prevents from storing true binary data in the database.

Will remove and use .encode("utf-8) & .decode(utf-8).

Reply:
Using the `encoding` instead of charset would not solve the problem?
```suggestion
        host='redis', port=6379, db=0, encoding="utf-8", decode_responses=True)
```

Reply:
@gyorb I think the response still will be decoded because of the `decode_responses=True` attribute.